CMakePresets.json

```diff
@@ -0,0 +1,136 @@
+{
+    "version": 3,
+    "cmakeMinimumRequired": {
+        "major": 3,
+        "minor": 21,
```

Comment:
This is a much higher version than what we require otherwise, do we really need it? I have no problem with this, just curious. FWIW [MSVS docs](https://learn.microsoft.com/en-us/cpp/build/cmake-presets-vs) only recommend requiring 3.14.

Reply:
This is independent of the version required to build. Version 3.21 is simply the first version to implement support for presets. (Technically it was 3.20 but the condition feature was added in 3.21)